test/facility-loan-transaction/get-loan-transaction-by-bundle-identifier.api-test.ts

```diff
@@ -0,0 +1,252 @@
+      },
+      SpreadRate: 0,
+      YearBasis: {
+        YearBasisCode: '',
+      },
+      IndexRateChangeFrequency: {
+        IndexRateChangeFrequencyCode: '',
+      },
+    });
+    const expectedFacilityLoanTransactionsWithAdditionalAccrual: GetFacilityLoanTransactionResponseItem = {
+      ...expectedLoanTransaction,
+      spreadRateCTL: 0,
+    };
+
+    givenAuthenticationWithTheIdpSucceeds();
+    requestToGetLoanTransaction().reply(200, loanTransactionInAcbsWithMoreThanOneCtlAccrual);
+
+    const { status, body } = await api.get(getLoanTransactionUrl);
+
+    expect(status).toBe(200);
+    expect(body).toStrictEqual(JSON.parse(JSON.stringify(expectedFacilityLoanTransactionsWithAdditionalAccrual)));
+  });
+
+  it(`returns a 400 response if the loan transaction is returned by ACBS and the 0th element of BundleMessageList is NOT a 'NewLoanRequest'`, async () => {
+    const invalidloanTransactionInAcbs = JSON.parse(JSON.stringify(loanTransactionInAcbs));
+    invalidloanTransactionInAcbs.BundleMessageList.unshift({
+      $type: 'AccrualScheduleAmountTransaction',
+    });
+
+    givenAuthenticationWithTheIdpSucceeds();
+    requestToGetLoanTransaction().reply(200, invalidloanTransactionInAcbs);
+
+    const { status, body } = await api.get(getLoanTransactionUrl);
+
+    expect(status).toBe(400);
+    expect(body).toStrictEqual({
+      statusCode: 400,
+      message: 'Bad request',
+      error: 'The provided bundleIdentifier does not correspond to a loan transaction.',
+    });
+  });
+
+  it('returns a 404 response if ACBS returns a 400 response with the string "BundleInformation not found"', async () => {
+    givenAuthenticationWithTheIdpSucceeds();
+    requestToGetLoanTransaction().reply(400, 'BundleInformation not found or user does not have access to it.');
+
+    const { status, body } = await api.get(getLoanTransactionUrl);
+
+    expect(status).toBe(404);
+    expect(body).toStrictEqual({
+      statusCode: 404,
+      message: 'Not found',
+    });
+  });
+
+  it('returns a 500 response if ACBS returns a 400 response without the string "BundleInformation not found"', async () => {
+    givenAuthenticationWithTheIdpSucceeds();
+    requestToGetLoanTransaction().reply(400, 'An error message from ACBS.');
+
+    const { status, body } = await api.get(getLoanTransactionUrl);
+
+    expect(status).toBe(500);
+    expect(body).toStrictEqual({
+      statusCode: 500,
+      message: 'Internal server error',
+    });
+  });
+
+  it('returns a 500 response if getting the loan transaction from ACBS returns a status code that is NOT 200 or 400', async () => {
+    givenAuthenticationWithTheIdpSucceeds();
+    requestToGetLoanTransaction().reply(401);
+
+    const { status, body } = await api.get(getLoanTransactionUrl);
+
+    expect(status).toBe(500);
+    expect(body).toStrictEqual({
+      statusCode: 500,
+      message: 'Internal server error',
+    });
+  });
```

Comment:
Don't we usually have a test that we return a 500 if ACBS returns a 400 where the response body is not a string?

Reply:
Good point, added.